src/components/modules/ModalWrapper.tsx

```diff
@@ -2,30 +2,28 @@
 
 import useModalStore from '@/store/modalStore';
 import { XCircleIcon } from '@heroicons/react/24/outline';
-import UploadFileForm from '@/components/widgets/modals/UploadFileForm';
-import DeleteTrackModal from '@/components/widgets/modals/DeleteTrackModal';
-import MultiDeleteModal from '@/components/widgets/modals/MultiDeleteModal';
-import TrackForm from '../widgets/modals/TrackForm';
 import { isAudioInfo, isStringArray, isTrack, ModalStore } from '@/types';
-import { JSX } from 'react';
+import { JSX, useMemo } from 'react';
+import dynamic from 'next/dynamic';
+
+const TrackForm = dynamic(() => import('../widgets/modals/TrackForm'));
+const UploadFileForm = dynamic(() => import('@/components/widgets/modals/UploadFileForm'));
+const DeleteTrackModal = dynamic(() => import('@/components/widgets/modals/DeleteTrackModal'));
+const MultiDeleteModal = dynamic(() => import('@/components/widgets/modals/MultiDeleteModal'));
 
 export default function ModalWrapper() {
   const closeModal = useModalStore((state) => state.closeModal);
   const modalType = useModalStore((state) => state.type);
   const defaults = useModalStore((state) => state.info);
 
 
-  const modalComponents: Record<string, JSX.Element | null> = {
-    create: <TrackForm type="create" defaults={null} />,
-    edit: isTrack(defaults) ? (
-      <TrackForm type="edit" defaults={defaults} />
-    ) : null,
-    file: isAudioInfo(defaults) ? <UploadFileForm defaults={defaults} /> : null,
-    delete: isTrack(defaults) ? <DeleteTrackModal defaults={defaults} /> : null,
-    deleteMulti: isStringArray(defaults) ? (
-      <MultiDeleteModal defaults={defaults} />
-    ) : null,
-  };
+ const modalComponents: Record<string, (() => JSX.Element | null) | undefined> = useMemo(() => ({
```

Comment:
```suggestion
 const modalComponents: Record<ModalType, (() => JSX.Element | null) | undefined> = useMemo(() => ({
```
Можна більш точно вказати які в тебе модалки є